spec/git-version-spec.cr

```diff
@@ -505,4 +505,23 @@ describe GitVersion do
       tmp.cleanup
     end
   end
-end
\ No newline at end of file
+
+  it "should properly manage prefixes" do
+    tmp = InTmp.new
+
+    begin
+      git = GitVersion::Git.new("dev", "master", tmp.@tmpdir, "v")
+
+      tmp.exec %(git init)
+      tmp.exec %(git checkout -b master)
+      tmp.exec %(git commit --no-gpg-sign --allow-empty --no-gpg-sign -m "1")
+      tmp.exec %(git tag "v1.0.0")
```

Comment:
we need tests for "mixed" situations e.g.:
 - everything works as expected switching from plain versioning to prefix
 - everything works as expected from prefixed tags to plain tags
 - tags with ""just the prefix" present
 - should we block/avoid any kind of prefix pattern?